### PR TITLE
Pull-in 2.3.0-beta1 release work

### PR DESCRIPTION
**Pull request by toru**
First step towards cutting a 2.3.0-beta2 release 🚢

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,10 +27,10 @@ PROJECT(h2o)
 SET(VERSION_MAJOR "2")
 SET(VERSION_MINOR "3")
 SET(VERSION_PATCH "0")
-SET(VERSION_PRERELEASE "-DEV")
+SET(VERSION_PRERELEASE "-beta1")
```
Comment (kazuho) [inline CMakeLists.txt]: This needs to be kept as "-DEV".

File: CMakeLists.txt [hunk 2]
```diff
@@ -27,10 +27,10 @@ PROJECT(h2o)
...
 SET(VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}${VERSION_PRERELEASE}")
 SET(LIBRARY_VERSION_MAJOR "0")
-SET(LIBRARY_VERSION_MINOR "14")
+SET(LIBRARY_VERSION_MINOR "15")
```
Comment (kazuho) [inline CMakeLists.txt]: This needs to be set to 16, because we are working for the release _after_ 2.3.0-beta1.

File: Changes [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Revision history for H2O.
 
-2.3.0-beta1 TBD
+2.3.0-beta1 2018-06-02 05:00:00+0000
```
Comment (kazuho) [inline Changes]: LGTM.

File: include/h2o/version.h [hunk 4]
```diff
@@ -27,17 +27,17 @@
 #endif
 
 #ifdef H2O_GITREV
-#define H2O_VERSION "2.3.0-DEV@" H2O_TO_STR(H2O_GITREV)
+#define H2O_VERSION "2.3.0-beta1@" H2O_TO_STR(H2O_GITREV)
```
Comment (kazuho) [inline include/h2o/version.h]: This file is automatically generated using the values of CMake, hence we do not need to do anything.

Comment (toru) [reply include/h2o/version.h]: Understood. I re-generated this file with the updated version values that you pointed out.

**Comment by toru** [comment]
Thank you for the review and the versioning walkthrough! -- The changeset is now tiny in respect to `master` with just the library version bump and the timestamp on the Changes file.

**Comment by kazuho** [comment]
Oh my bad, I should have noticed and said that the v2.3.x branch should not be used as a workplace for composing a pull request - it's not a branch that should contain code we ship, like master.

OTOH, the changes being proposed look good to me. Thank you for working on them. Considering the above point (i.e., v.2.3.x is not a WIP), I am going to squash-merge this.

**Pull request merged**